src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -0,0 +1,248 @@
+
+        attendanceAssignment2 = new AttendanceAssignment();
+        attendanceAssignment2.setAssignmentId(ASSIGNMENT_ID_2);
+        attendanceAssignment2.setAssignmentName(ASSIGNMENT_NAME_2);
+        attendanceAssignment2.setCanvasAssignmentId(CANVAS_ASSIGNMENT_ID);
+        attendanceAssignment2.setGradingOn(true);
+        attendanceAssignment2.setAssignmentPoints(ASSIGNMENT_POINTS);
+        attendanceAssignment2.setAbsentPoints(0.0);
+        attendanceAssignment2.setExcusedPoints(0.0);
+        attendanceAssignment2.setTardyPoints(0.0);
+        attendanceAssignment2.setPresentPoints(100.0);
+        attendanceAssignment2.setStatus(AttendanceAssignment.Status.UNKNOWN);
+
+        assignment = new Assignment();
+        assignment.setPointsPossible(ASSIGNMENT_POINTS);
+        assignmentOptional = Optional.of(assignment);
+
+        course = new AttendanceCourse();
+        course.setCanvasCourseId(COURSE_ID);
+        course.setAttendanceType(AttendanceType.SIMPLE);
+        course.setShowNotesToStudents(true);
+
+        section1 = new AttendanceSection();
+        section1.setSectionId(SECTION_1_ID);
+        section1.setCanvasCourseId(COURSE_ID);
+        section1.setCanvasSectionId(SECTION_1_ID);
+        section1.setName(SECTION_1_NAME);
+
+        section2 = new AttendanceSection();
+        section2.setSectionId(SECTION_2_ID);
+        section2.setCanvasCourseId(COURSE_ID);
+        section2.setCanvasSectionId(SECTION_2_ID);
+        section2.setName(SECTION_2_NAME);
+
+        studentCommentsMap1 = new HashMap<>();
+        studentCommentsMap1.put(STUDENT_1_ID, STUDENT_1_COMMENT);
+        studentCommentsMap1.put(STUDENT_2_ID, STUDENT_2_COMMENT);
+
+        studentCommentsMap2 = new HashMap<>();
+        studentCommentsMap2.put(STUDENT_3_ID, STUDENT_3_COMMENT);
+        studentCommentsMap2.put(STUDENT_4_ID, STUDENT_4_COMMENT);
+
+        progress = new Progress();
+        progress.setId(PROGRESS_ID);
+        progress.setUrl(CANVAS_URL);
+        progress.setWorkflowState("queued");
+        progressOptional = Optional.of(progress);
+
+        Whitebox.setInternalState(assignmentSubmitter, "canvasAssignmentAssistant", canvasAssignmentAssistant);
+        Whitebox.setInternalState(assignmentSubmitter, "assignmentService", assignmentService);
+        Whitebox.setInternalState(assignmentSubmitter, "sectionService", sectionService);
+        Whitebox.setInternalState(assignmentSubmitter, "attendanceCourseService", attendanceCourseService);
+        Whitebox.setInternalState(assignmentSubmitter, "attendanceService", attendanceService);
+        Whitebox.setInternalState(assignmentSubmitter, "canvasApiWrapperService", canvasApiWrapperService);
+        Whitebox.setInternalState(assignmentSubmitter, "assignmentValidator", assignmentValidator);
+
+
+
+    }
+
+    @Test
+    public void submitCourseAttendancesHappyPath() throws IOException, AttendanceAssignmentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.AttendanceAssignmentException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)